static/js/components/Payment.js

```diff
@@ -19,11 +21,68 @@ export default class Payment extends React.Component {
     payment: RestState,
     payableKlassesData: Array<Object>,
     selectedKlass: Object,
+    now: moment$Moment,
     sendPayment: () => void,
     setPaymentAmount: (event: InputEvent) => void,
     setSelectedKlassKey: (event: InputEvent) => void
   };
 
+  getTotalOwedUpToInstallment = (nextInstallmentIndex: number): number => {
+    const {
+      selectedKlass: { installments }
+    } = this.props;
+
+    return R.compose(
+      R.sum,
+      R.map(R.prop('amount')),
+      R.slice(0, nextInstallmentIndex + 1)
+    )(installments);
+  };
+
+  getDeadlineDates = R.map(
+    R.compose(
+      moment,
+      R.prop('deadline')
+    )
+  );
+
+  hasMissedPreviousInstallment = (previousInstallmentIndex: number): boolean => {
+    const { selectedKlass } = this.props;
+
+    return previousInstallmentIndex >= 0 &&
+      selectedKlass.total_paid < this.getTotalOwedUpToInstallment(previousInstallmentIndex);
+  };
+
+  getPaymentDeadlineText = (): string => {
+    const { selectedKlass, now } = this.props;
+
+    let installments = selectedKlass.installments;
+    let installmentDeadlineText = '',
+      finalInstallmentDeadline = '';
+
+    if (installments.length > 1) {
+      let deadlineDates = this.getDeadlineDates(installments);
+      let nextInstallmentIndex = R.findIndex(R.lt(now), deadlineDates);
+      if (nextInstallmentIndex !== installments.length - 1) {
+        let totalOwedForInstallment = this.getTotalOwedUpToInstallment(nextInstallmentIndex);
+        if (selectedKlass.total_paid < totalOwedForInstallment) {
+          installmentDeadlineText =
+            `A total downpayment of ${formatDollarAmount(totalOwedForInstallment)} ` +
+            `is due ${formatReadableDate(deadlineDates[nextInstallmentIndex])}.`;
+        }
+      }
+      if (this.hasMissedPreviousInstallment(nextInstallmentIndex - 1)) {
```

Comment:
Suggestion: have this method parameterized just on `nextInstallmentIndex` and do the index math inside it since the concept of previous is relative to that.